src/test/java/seedu/address/model/person/FilterBuyerByCharacteristicsPredicateTest.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.characteristics.Characteristics;
+import seedu.address.testutil.PersonBuilder;
+
+public class FilterBuyerByCharacteristicsPredicateTest {
+
+    @Test
+    public void equals() {
+        Characteristics oneCharacteristic = new Characteristics("HDB");
+        Characteristics twoCharacteristic = new Characteristics("HDB;5-room");
+
+        FilterBuyerByCharacteristicsPredicate firstPredicate =
+                new FilterBuyerByCharacteristicsPredicate(oneCharacteristic);
+        FilterBuyerByCharacteristicsPredicate secondPredicate =
+                new FilterBuyerByCharacteristicsPredicate((twoCharacteristic));
+
+        //same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        //same value -> returns true
+        FilterBuyerByCharacteristicsPredicate firstPredicateCopy =
+                new FilterBuyerByCharacteristicsPredicate(oneCharacteristic);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        //different types -> returns false
+        assertFalse(firstPredicate.equals("HDB"));
+
+        //null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        //different characteristics -> false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_buyerContainsCharacteristics_returnTrue() {
+        //One characteristics
+        FilterBuyerByCharacteristicsPredicate predicate =
+                new FilterBuyerByCharacteristicsPredicate(new Characteristics("Bishan"));
+        assertTrue(predicate.test(new PersonBuilder().withDesiredCharacteristics("Bishan").build()));
+
+        // Contains one characteristics
+        predicate = new FilterBuyerByCharacteristicsPredicate(new Characteristics("HDB"));
+        assertTrue(predicate.test(new PersonBuilder().withDesiredCharacteristics("HDB ; 5-room").build()));
+
+        // Only match one characteristics
+        predicate = new FilterBuyerByCharacteristicsPredicate(new Characteristics("HDB ; 5-room"));
+        assertTrue(predicate.test(new PersonBuilder().withDesiredCharacteristics("HDB").build()));
```

Comment:
Actually should this be true? If the user is filtering by 2 characteristics, we shouldn't be returning person that only fulfils one of the characteristic? 

Might need to change `containsAnyGivenCharacteristics` to `containsAllGivenCharacteristics` if this is the case

Reply:
Yup I think false would make more sense too